File: WoofIOS/SwiftPackages/Data/Repositories/Sources/Repositories/Dogs/Mapper/DogSizeDTOToDomainMapper.swift
import DI
import DTO
import Models

@Factory
public final class DogSizeDTOToDomainMapper {
    public func callAsFunction(_ size: String) -> DogSize {
        switch size {
        case DogSizeDTO.small: .small
        case DogSizeDTO.large: .large
        default: .medium
        }
    }
}

File: WoofAndroid/data/repositories/src/main/kotlin/com/perrystreet/woof/repositories/dogs/mapper/DogSizeDTOToDomainMapper.kt
package com.perrystreet.woof.repositories.dogs.mapper

import com.perrystreet.woof.dto.dog.DogSizeDTO
import com.perrystreet.woof.models.dog.DogSize
import org.koin.core.annotation.Factory

@Factory
class DogSizeDTOToDomainMapper {
    operator fun invoke(size: String): DogSize = when (size) {
        DogSizeDTO.Small -> DogSize.Small
        DogSizeDTO.Large -> DogSize.Large
        else -> DogSize.Medium
    }
}
